src/views/instances/list/filter-deployments.tsx

```diff
@@ -7,17 +7,34 @@ const useDeployments = () => {
     const [{ data, error, loading }, refetch] = useAuthAxios<GroupWithDeployments[]>('/deployments', {
         useCache: false,
     })
+
     const [showOnlyMyInstances, setShowOnlyMyInstances] = useState(false)
     const { currentUser } = useContext(AuthContext)
 
-    const filterDeploymentsByUser = useMemo(() => {
-        return data?.map((group) => ({
+    useEffect(() => {
+        const savedShowOnlyMyInstances = localStorage.getItem('showOnlyMyInstances')
+        if (savedShowOnlyMyInstances !== null) {
+            setShowOnlyMyInstances(JSON.parse(savedShowOnlyMyInstances))
+        }
+    }, [])
```

Comment:
`useEffect` is not really needed here - and in my opinion is better done inside `useState(() => { }`-initial callback.
I would've also have added `try-catch` and fallback to false

```
    const [showOnlyMyInstances, setShowOnlyMyInstances] = useState(() => {
        try {
            return JSON.parse(localStorage.getItem(LOCAL_STORAGE_KEY))
        }
        catch {
            return false
        }
    })
```